### PR TITLE
HADOOP-17936. Fix test failure after reverting HADOOP-16878 from branch-3.3

### PR DESCRIPTION
**Pull request by sunchao**
HADOOP-17936
### Description of PR

After reverting HADOOP-16878 from branch-3.3, test `TestLocalFSCopyFromLocal.testDestinationFileIsToParentDirectory` started to fail since it depends on the behavior change in the JIRA. This fixed the test.

### How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by sunchao** [comment]
cc @bgaborg @steveloughran

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   9m 54s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 1 new or modified test files.  |
|||| _ branch-3.3 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  33m 51s |  |  branch-3.3 passed  |
| +1 :green_heart: |  compile  |  18m 18s |  |  branch-3.3 passed  |
| +1 :green_heart: |  checkstyle  |   0m 50s |  |  branch-3.3 passed  |
| +1 :green_heart: |  mvnsite  |   1m 31s |  |  branch-3.3 passed  |
| +1 :green_heart: |  javadoc  |   1m 31s |  |  branch-3.3 passed  |
| +1 :green_heart: |  spotbugs  |   2m 22s |  |  branch-3.3 passed  |
| +1 :green_heart: |  shadedclient  |  25m 58s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 56s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  17m 35s |  |  the patch passed  |
| +1 :green_heart: |  javac  |  17m 35s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   0m 46s | [/results-checkstyle-hadoop-common-project_hadoop-common.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3478/1/artifact/out/results-checkstyle-hadoop-common-project_hadoop-common.txt) |  hadoop-common-project/hadoop-common: The patch generated 2 new + 1 unchanged - 1 fixed = 3 total (was 2)  |
| +1 :green_heart: |  mvnsite  |   1m 28s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m 31s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   2m 30s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  26m 51s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  16m 45s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 48s |  |  The patch does not generate ASF License warnings.  |
|  |   | 162m 24s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3478/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/3478 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell |
| uname | Linux 53749a3faf37 4.15.0-142-generic #146-Ubuntu SMP Tue Apr 13 01:11:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | branch-3.3 / 2e3024bab21328c78bc9bb5f54f1c384ab2e6522 |
| Default Java | Private Build-1.8.0_292-8u292-b10-0ubuntu1~18.04-b10 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3478/1/testReport/ |
| Max. process+thread count | 1259 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common U: hadoop-common-project/hadoop-common |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3478/1/console |
| versions | git=2.17.1 maven=3.6.0 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Review by dongjoon-hyun — COMMENTED** [review]
Thank you, @sunchao

**Review by steveloughran — APPROVED** [review]
+1

this change is in trunk already?

**Comment by sunchao** [comment]
@steveloughran good question. it's not because we still have HADOOP-16878 / #1857 in the trunk so the test is passing there. Should we revert it there too? because the JIRA is re-opened.

**Review by ayushtkn — APPROVED** [review]
Thanx @sunchao, IMO we should revert HADOOP-16878 from trunk as well and push this to both branches.

**Comment by dongjoon-hyun** [comment]
Thank you guys. Could you merge this to branch-3.3 first, please?

**Comment by sunchao** [comment]
Sounds good @ayushtkn . I've reverted them from trunk too, and open #3482 to fix the test there.

**Comment by sunchao** [comment]
@dongjoon-hyun I'll wait until the fix is merged to trunk first, before merging this to branch-3.3

**Comment by dongjoon-hyun** [comment]
Got it~ No problem.

**Review by bgaborg — APPROVED** [review]

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 55s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  1s |  |  codespell was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 1 new or modified test files.  |
|||| _ branch-3.3 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  33m 54s |  |  branch-3.3 passed  |
| +1 :green_heart: |  compile  |  18m  3s |  |  branch-3.3 passed  |
| +1 :green_heart: |  checkstyle  |   0m 48s |  |  branch-3.3 passed  |
| +1 :green_heart: |  mvnsite  |   1m 30s |  |  branch-3.3 passed  |
| +1 :green_heart: |  javadoc  |   1m 32s |  |  branch-3.3 passed  |
| +1 :green_heart: |  spotbugs  |   2m 22s |  |  branch-3.3 passed  |
| +1 :green_heart: |  shadedclient  |  26m 35s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 57s |  |  the patch passed  |
| +1 :green_heart: |  compile  |  17m 36s |  |  the patch passed  |
| +1 :green_heart: |  javac  |  17m 36s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 47s |  |  hadoop-common-project/hadoop-common: The patch generated 0 new + 1 unchanged - 1 fixed = 1 total (was 2)  |
| +1 :green_heart: |  mvnsite  |   1m 29s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m 30s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   2m 31s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  26m 44s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  16m 54s |  |  hadoop-common in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 48s |  |  The patch does not generate ASF License warnings.  |
|  |   | 153m 57s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3478/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/3478 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell |
| uname | Linux 7ee738ceaa6e 4.15.0-142-generic #146-Ubuntu SMP Tue Apr 13 01:11:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | branch-3.3 / b649c3c5ef1b290fc84f5dd161e808f1e29d2ab6 |
| Default Java | Private Build-1.8.0_292-8u292-b10-0ubuntu1~18.04-b10 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3478/2/testReport/ |
| Max. process+thread count | 2661 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common U: hadoop-common-project/hadoop-common |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-3478/2/console |
| versions | git=2.17.1 maven=3.6.0 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0-SNAPSHOT https://yetus.apache.org |


This message was automatically generated.

**Comment by sunchao** [comment]
Thanks all! I just merged this to branch-3.3

**Pull request merged**
